integration_testing/tests/0-startup.int.js

```diff
@@ -0,0 +1,170 @@
+const { percySnapshot } = require('../helpers/percy');
+
+describe('Kolibri setup wizard', () => {
+  beforeAll(() => {
+    browser.url('/');
+  });
+
+  const totalSteps = 7;
+  const stepTitle = stepName => `${stepName} of ${totalSteps}`;
+  const assertStepTitle = stepName => {
+    expect($('.ui-toolbar .ui-toolbar__body').getText()).toEqual(stepTitle(stepName));
```

Comment:
are these assertions coming from jest, webdriver, percy, or elsewhere?

Reply:
I'm not sure exactly what you're asking. Here's more info:
- I created the `assert*` functions
- `expect()` is provided by jasmine
- `$` and `$$` are globals from methods on `browser` (`browser.$ === $`) and are provided by webdriverio
-  webdriver and webdriverio are two different things

Reply:
cool, thanks for the info.

I was mostly wondering whether this overlaps with some of what we currently use Jest and the [Vue test utils](https://vue-test-utils.vuejs.org/) for? I.e. the `describe`...`expect` pattern looks a lot like some of our [existing tests](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/test/views/elapsed-time.spec.js#L24-L29).

Reply:
Ah yeah, a bunch of the JS test runners allow for the BDD style. Jest actually uses a forked version of Jasmine internally. The default runner for Jest is their Jasmine fork.

Since Jest is a full-feature suite, it includes everything you'd need by default, which includes `jsdom`, a virtual DOM. That's not really compatible with running integration tests. `jest-puppeteer` has its own custom environment based of Jest's Node environment. No-one has made a Jest runner for `webdriverio` by creating a custom environment, although there has been a lot of discussion and interest. If `jest-puppeteer` can do it, it may be possible to create a jest runner for `webdriverio` that makes everything happy. That would be an awesome project that would get a lot of visibility.

Reply:
super helpful, thanks